### ISSUE TITLE
[QUESTION] How to save out online Soil and Fertilizer NOx emission diagnostic

### ISSUE THREAD

**Issue by xpye98** [geoschem/geos-chem#433]
# Ask a question about GEOS-Chem:

Hi,
I'm using the latest version 12.9.3.

In new version, the default soil NOx emission is offline, but it cannot separate natural and fertilizer source (since the data only calculate overall soil NOx). So, here I still choose online soil NOx, like versions before.The problem is, how should I set HEMCO_Diagn.rc file?

When using offline soil NOx, it looks like this:
`
EmisNO_Soil        NO     0      3   -1   2   kg/m2/s  NO_emission_flux_from_soil
`
Now using online, I suppose it should be:
`EmisNO_Soil        NO     104    -1  -1   2   kg/m2/s  NO_emission_flux_from_soil`
`EmisNO_Fert        NO     104    -1  -1   2   kg/m2/s  NO_emission_flux_from_fertilizer_only`

The Extnum should be changed from 0 (base emission) to 104 (online soil NOx emission), but how to set `cat` and `hier` ? If I set the numbers like above, `EmisNO_soil` and `EmisNO_Fert` return the same result.

Thanks,
Xingpei

**Comment by msulprizio** [comment]
It looks like you are doing this correctly. You should be able to save out fertilizer emissions using: 

```
EmisNO_Soil        NO     104    -1  -1   2   kg/m2/s  NO_emission_flux_from_soil_and_fertilizer
EmisNO_Fert        NO     104    -1  -1   2   kg/m2/s  NO_emission_flux_from_fertilizer_only
```

Have you tried differencing the two values? They may look identical, but subtracting EmisNO_Fert from EmisNO_Soil should return the value of NO emissions from soil only. See commits https://github.com/geoschem/geos-chem/commit/f083b7a6d3f9a0bf988d9a00d57a03bbe763a247 and https://github.com/geoschem/geos-chem-unittest/commit/0a4ff6c71d564b5c720d57872348f01eafb52d5a for more details.

Another test may be to run simulations with and without fertilizer NOx turned on in the SoilNOx extension and look at the difference between the two soil NOx emissions output. This is controlled by a switch in HEMCO_Config.rc:

```
104     SoilNOx                : on    NO
    --> Use fertilizer NOx     :       true
```

**Comment by xpye98** [comment]
> It looks like you are doing this correctly. You should be able to save out fertilizer emissions using:

Thanks for replying! 
I tried with and without fertilizer NOx, `EmisNO_soil` and `EmisNO_Fert` still return the same result.

I even did this change in Code.12.9.3/HEMCO/src/Extensions/hcox_soilnox_mod.F90 (about line 572) to test
![image](https://user-images.githubusercontent.com/59330122/92502282-3a61dc80-f232-11ea-97f8-94cf7ca49b8f.png)
Here, FLUX_2D is overall soil NOx, DIAG is fertilizer NOx only. So changes here should cut half total soil NOx but remain fertilizer emission unchanged. However, `EmisNO_soil` and `EmisNO_Fert` in diagnostic both decrease 50%!

So, I wonder, does model really recognize soil and fert just from their name? (but as I see in code, it really does)
Maybe we should change category and hierarchy value? I don't know...

**Comment by xpye98** [comment]
> It looks like you are doing this correctly. You should be able to save out fertilizer emissions using:

Hi, 
I add these lines in hcox_soilnox_mod.F90 to debug (already recompiled), and find that the Logic variable `DoDiagn` doesn't change, which means the model cannot tell the 'EmisNO_Fert'.
![image](https://user-images.githubusercontent.com/59330122/92894325-0dd5dc80-f44d-11ea-9f47-d4920564af1c.png)

I tried to manually set `DoDiagn` to True, and print out FLUX_2D (soil emissions) and DIAG (fert emissions) to screen. They both make sense. However, the results in HEMCO_diagnostic file are strill wrong, EmisNO_Fert are even larger than EmisNO_Soil.

Here, I use geosfp 4x5 benchmark simulation, and I followed the changes in [geoschem/geos-chem-unittest@0a4ff6c](https://github.com/geoschem/geos-chem-unittest/commit/0a4ff6c71d564b5c720d57872348f01eafb52d5a).

**Comment by xpye98** [comment]
Here is the sum of the problem I encounter (people in my group also share the same problem):

First, in hcox_soilnox_mod.F90 file, the logical variable `DoDiagn` is always false, even if I add EmisNO_Fert into HEMCO_Diagn.rc.

Second, after manually setting `DoDiagn` to True, EmisNO_Fert would be more than EmisNO_Soil ( if `DoDiagn` is false, they are the same). I debugged the hcox_soilnox_mod.F90 and found that this is because the function `HCO_EmisAdd` assigns `FLUX_2D`(which is soil NOx emission) to both EmisNO_Soil and EmisNO_Fert. And then down below, `DIAG`(fertilizer NOx emission) is assigned to EmisNO_Fert, which results overcalculated EmisNO_Fert.

**Comment by msulprizio** [comment]
It appears the capability to save out fertilizer NOx emissions separately was inadvertently removed in GEOS-Chem 12.7.0. In that version, a majority of the [code for bpch diagnostic was removed](https://github.com/geoschem/geos-chem/issues/82) and this included code for creating manual HEMCO diagnostics in `hcoi_gc_diagn_mod.F90`. In that module, code for creating the `EmisNO_Fert` diagnostic was also removed but is still needed. I am working on a fix to create and update `EmisNO_Fert` directly in `hcox_soil_nox_mod.F90`.

**Comment by msulprizio** [comment]
This is now resolved in https://github.com/geoschem/HEMCO/commit/be2f114fe11363f522141aa82072cbcc5bd22744 and will be included in GEOS-Chem 13.0.0.
